### PR TITLE
Devmapper Graph Driver Misc cleanup

### PR DESCRIPTION
**Pull request by rhvgoyal**
This pull request fixes issue #12032

### CODE REVIEW

**Comment by rhvgoyal** [comment]
@vbatts 

Can you please have a look.

**Comment by vbatts** [comment]
One not, but it all LGTM and a nice cleanup

**Comment by vbatts** [comment]
*nit

**Comment by rhvgoyal** [comment]
@vbatts 

Fixed the issue you had pointed out.

File: daemon/graphdriver/devmapper/deviceset.go [hunk 1]
```diff
@@ -723,14 +727,22 @@ func setCloseOnExec(name string) {
...
 	// FIXME(vbatts) push this back into ./pkg/devicemapper/
-	logrus.Debugf("libdevmapper(%d): %s:%d (%d) %s", level, file, line, dmError, message)
+	if level <= devicemapper.LogLevelErr {
+		logrus.Errorf("libdevmapper(%d): %s:%d (%d) %s", level, file, line, dmError, message)
+	} else if level <= devicemapper.LogLevelInfo {
+		logrus.Infof("libdevmapper(%d): %s:%d (%d) %s", level, file, line, dmError, message)
+	} else {
+		// FIXME(vbatts) push this back into ./pkg/devicemapper/
+		logrus.Debugf("libdevmapper(%d): %s:%d (%d) %s", level, file, line, dmError, message)
+	}
```
Comment (icecrime) [inline daemon/graphdriver/devmapper/deviceset.go]: Any reason not to map `devicemapper.LogLevelWarn` to `logrus.Warnf`?

File: pkg/devicemapper/devmapper.go [hunk 2]
```diff
@@ -284,9 +284,9 @@ func FindLoopDeviceFor(file *os.File) *os.File {
 	return nil
 }
 
-func UdevWait(cookie uint) error {
-	if res := DmUdevWait(cookie); res != 1 {
-		logrus.Debugf("Failed to wait on udev cookie %d", cookie)
+func UdevWait(cookie *uint) error {
```
Comment (icecrime) [inline pkg/devicemapper/devmapper.go]: Sorry, what's the rationale for the `*uint`?

Comment (crosbymichael) [inline pkg/devicemapper/devmapper.go]: It looks like this method is being called in a defer and the value for cookie is not yet set when the defer is initialized

Comment (rhvgoyal) [inline pkg/devicemapper/devmapper.go]: Yes. That's the reason. We want to make sure we get the final value of cookie (as set by libdm calls) in deferred function.

File: daemon/graphdriver/devmapper/deviceset.go (hunk 1 above)
Comment (rhvgoyal) [inline daemon/graphdriver/devmapper/deviceset.go]: No. It just did not occur to me. I will change it to map LogLevelWarn to logrus.Warnf.

File: pkg/devicemapper/devmapper.go (hunk 2 above)
Comment (icecrime) [inline pkg/devicemapper/devmapper.go]: I see, sorry!

**Comment by jfrazelle** [comment]
-96 awesomeee

**Comment by vbatts** [comment]
@jfrazelle what are you even saying? ;-)

**Comment by vbatts** [comment]
LGTM

**Comment by jfrazelle** [comment]
to the line of code, theres minus a lot hahaha LGTM

**Pull request merged**